feat: Implement Feeless Operations for Gatekeeper and Miners in Task Management

Nathy-bajo opened the pull request:
Feeless Task Scheduling for Gatekeeper:

- Added Gatekeeper type to pallet_task_management config

- Modified task_scheduler to be feeless only for the gatekeeper account

- Implemented gatekeeper account in runtime using parameter_types!

Miner-Only Functions with Feeless Execution:

- Added miner checks to all miner-called functions (submit_completed_task, verify_completed_task, resolve_completed_task)

- Made all miner functions feeless (Pays::No)

- Ensured only registered miners can call these functions

Runtime Configuration:

- Added gatekeeper account configuration

- Updated mock runtime for testing

Edge-connect:
- Added new events for when a worker is penalized or suspended.
- Added helper function to check worker's status.
- Added extrinsic to penalize workers.

Worker:
- Updated `WorkerReputation`

Task-management: 
- Modified task_scheduler to check if worker's status and reputation.
- Added a check for worker's existence.
- Added check_rate_limit logic.

Ronnie-Ahmed reviewed (changes requested):
Kindly update the payment pallet, as the implementation has been revised. Please make changes only to `payment/src/lib.rs` and `payment/src/test.rs`. Additionally, add the required files to `primitives/src/lib.rs` and include the new module in `primitives/src/payment.rs`. Feel free to reach out if you need any clarification.

Nathy-bajo commented:
As requested by @tom-blk I updated the codebase to make `task_scheduler` feeless only for a designated `gatekeeper account` (while regular users still pay fees) and modified miner-specific functions (submit_completed_task, verify_completed_task, resolve_completed_task) to be feeless exclusively for `registered miners`, with strict checks ensuring only `miners` can call them, optimizing gas efficiency and economic incentives for network participants.

Nathy-bajo commented:
> Kindly update the `payment` pallet according to the latest changes on the master branch.

Updated the latest changes to the master branch.

> Please make changes only to `payment/src/lib.rs` and `payment/src/test.rs`. Additionally, add the required files to `primitives/src/lib.rs` and include the new module in `primitives/src/payment.rs`. 

What additional required files do you mean, please?

Ronnie-Ahmed commented:
All necessary components for the payment pallet have been implemented as required. You do not need to add anything additional .

Nathy-bajo commented:
@tom-blk @Ronnie-Ahmed Pease review.
I also updated some deprecated codebase

tom-blk reviewed (commented):
I think this looks good, but instead of the runtime defining the gatekeeper account it should be set by a sudo call. Also we will need a mechanism to prevent the scenario if the owner of the miner is malicious, cracks it and starts DoS-ing the parachain, do you have any ideas regarding that?

tom-blk reviewed (approved):
Good work, will merge!
